From 28b25d7c5eee288a8fda2bfe0246baa0cee8cb59 Mon Sep 17 00:00:00 2001
From: Marek Marczykowski <marmarek@invisiblethingslab.com>
Date: Mon, 11 Jun 2012 22:49:31 +0200
Subject: [PATCH] pvops: respect 'removable' xenstore flag for block devices

Especially this is needed by pmount to allow mount qvm-block attached devices
by normal user.
---
 drivers/block/xen-blkfront.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/block/xen-blkfront.c b/drivers/block/xen-blkfront.c
index 9df516a56bb2..a9ba4956be22 100644
--- a/drivers/block/xen-blkfront.c
+++ b/drivers/block/xen-blkfront.c
@@ -2341,6 +2341,7 @@ static void blkfront_connect(struct blkfront_info *info)
 	char *envp[] = { "RESIZE=1", NULL };
 	int err, i;
 	struct blkfront_ring_info *rinfo;
+	int removable;
 
 	switch (info->connected) {
 	case BLKIF_STATE_CONNECTED:
@@ -2408,6 +2409,12 @@ static void blkfront_connect(struct blkfront_info *info)
 		}
 	}
 
+	err = xenbus_gather(XBT_NIL, info->xbdev->otherend,
+			    "removable", "%d", &removable,
+			    NULL);
+	if (!err && removable)
+		binfo |= VDISK_REMOVABLE;
+
 	err = xlvbd_alloc_gendisk(sectors, info, binfo, sector_size,
 				  physical_sector_size);
 	if (err) {
-- 
2.25.1

